docs/design/proposals/server-side-override/server-side-override.md

```diff
@@ -0,0 +1,224 @@
+# ACK Handling Server Side Defaults
+
+## Problem
+
+During reconciliation, ACK service controller consumes the custom resource spec (i.e. desired state) from “etcd” and
+then creates/updates the AWS resource to match that desired state. ACK service controller uses the custom resource spec
+fields to construct Create/Update AWS API request.
+
+However, some AWS API’s behavior does not completely rely on the fields present in Create/Update AWS API request alone.
+In the backend, these AWS APIs sometime update the value of field present in request
+with newly calculated value which represents the latest state of the resource in AWS. AWS APIs return these updated 
+values with Create/Update API output and Read/List/Describe API output after resource creation.
+
+Following problem arise due to this behavior of AWS APIs,
+
+* When an AWS API does not strictly use the property value as is in the backend, but updates it, the desired spec of
+the K8s resource in the “etcd” does not correctly represent the actual state of resource in AWS. This is a problem because
+for K8s user, the spec of custom resource is the source of truth for state of K8s resource.
+
+
+## Solution Requirements
+
+1. Consistency with K8s conventions
+2. Consistency across ACK service controllers
+3. GitOps compatibility
+4. User experience for custom resource lifecycle 
+
+## Solution
+
+In this solution, If an existing value is updated as part of Create/Update API output, ACK service controller will persist 
```

Comment:
I know this may be obvious to some, but it may be useful to make explicit why the annotation/mapping is only written during Creates/Updates, and not during Reads